### PR TITLE
telemetry: Add SQL statistics bucket into telemetry data

### LINKED ISSUE #25005: Add slow query statistic information into telemetry data
## Development Task
As title, implement this feature.

### PR DESCRIPTION
**Pull request by YinWeiling**
### What problem does this PR solve?

Issue Number: close #25005 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
add SQL statistics bucket to telemetry report data

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)
Setup TiDB
Create a tidb_ops_type table with type
Execute: admin show telemetry.

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- add SQL statistics bucket to telemetry report data

### CODE REVIEW

File: telemetry/data_window.go [hunk 1]
```diff
@@ -117,6 +362,12 @@ func getWindowData() []*windowData {
 	i := 0
 	for i < len(rotatedSubWindows) {
 		thisWindow := *rotatedSubWindows[i]
+		var startWindow windowData
+		if i == 0 {
+			startWindow = thisWindow
+		} else {
+			startWindow = *rotatedSubWindows[i-1]
+		}
```
Comment (xhebox) [inline telemetry/data_window.go:370]: So it is iterating like `thisWindow = r[1] - r[0]`, `thisWindow = r[2] - r[0]`, `thisWindow = r[3] - r[1]` for `thisWindow.SQLUsage.SQLTotal = rotatedSubWindows[i].SQLUsage.SQLTotal - startWindow.SQLUsage.SQLTotal`?

I don't quite understand what you want to do, but is the distance of `2` between windows intended?

**Review by xhebox — COMMENTED** [review]
A little bit long, may be you can reduce the length of PR by some refactors.

File: domain/domain.go [hunk 2]
```diff
@@ -1067,6 +1067,7 @@ func (do *Domain) TelemetryReportLoop(ctx sessionctx.Context) {
 
 // TelemetryRotateSubWindowLoop create a goroutine that rotates the telemetry window regularly.
 func (do *Domain) TelemetryRotateSubWindowLoop(ctx sessionctx.Context) {
+	logutil.BgLogger().Info("InitSQLStats already")
```
Comment (qw4990) [inline domain/domain.go:1070]: Should we use debug level here?

Comment (qw4990) [reply domain/domain.go:1070]: Emm and this log message seems too casual, could you make it more formal? 🤔

**Comment by xhebox** [comment]
The issue is actually closed by #25006

**Comment by YinWeiling** [comment]
邮件已经收到啦～

**Pull request closed without merging**